pkg/adaptation/adaptation.go

```diff
@@ -208,6 +209,17 @@ func (r *Adaptation) RunPodSandbox(ctx context.Context, evt *StateChangeEvent) e
 	return r.StateChange(ctx, evt)
 }
 
+// UpdatePodSandbox relays the corresponding CRI request to plugins.
+func (r *Adaptation) UpdatePodSandbox(_ context.Context, _ *UpdatePodSandboxRequest) (*UpdatePodSandboxResponse, error) {
```

Comment:
After thinking more about this, I think this should be an event: unlike for containers, pod cgroup management is in the kubelet. Therefore there is not much point in allowing a plugin to respond with requested changes to the update. There is nothing the runtime could do to make that happen.

Reply:
Is there a likelihood that pod-level cgroup management could shift to the container runtime in the future? If so I think it would be more intuitive for this to not be just an event but to also have its own response type (which can be empty for now per your other feedback).

It looks like this method signature would fit in with [the approach in your branch](https://github.com/klihub/nri/commit/dfd20ac81115ba329c79e0e7d262d8a51448c087) so I'd be inclined to not make any changes here.

Reply:
Yes, I agree. I think we should stick to the normal RPC  signatures on the wire and just have the returned UpdatePodSandboxResponse struct be empty. In the stub/plugin interface (the handlers) though, it would probably be better to only have an returned error in the signature of the handler and add the the empty returned &UpdatePodSandboxResponse{} in the stub. WDYT ?

Reply:
> Is there a likelihood that pod-level cgroup management could shift to the container runtime in the future? If so I think it would be more intuitive for this to not be just an event but to also have its own response type (which can be empty for now per your other feedback).

I know some folks who'd like to at least give some serious consideration for that to happen in the long run.